### PR TITLE
Fix docstrings, pep8, pep257, and add param doc for in exceptions.py

### PR DESCRIPTION
**Pull request by terryjbates**

### CODE REVIEW

File: cookiecutter/exceptions.py [hunk 1]
```diff
@@ -1,103 +1,90 @@
+#!/usr/bin/env python
```
Comment (hackebrot) [inline cookiecutter/exceptions.py]: Can you please remove the shebang? Thank you! 🙇

**Comment by terryjbates** [comment]
@hackebrot Appears that removing the shebang line causes Travis CI to fail, due to an `InvocationError`

https://travis-ci.org/audreyr/cookiecutter/jobs/152247749

A bit mysterious. How shall we proceed?

**Comment by michaeljoseph** [comment]
@terryjbates the test failure looks unrelated to your change.

And @hackebrot has already raised an issue for the underlying problem: https://github.com/pytest-dev/pytest-mock/issues/59

I guess we should pin `pytest-mock` for now.

**Comment by michaeljoseph** [comment]
@terryjbates rebasing on master to get 2d0affe should fix this failure now.

**Comment by terryjbates** [comment]
Mkay...not sure if I have horked this up or not, so let me know if I have! Tried the following:

1. On local master branch executed `git fetch upstream` and `git merge upstream/master`.
2. Changed branches to the topic branch of `tbates-pep257-742-exceptions.`
3. Did `git pull.`
4. Did `git rebase -i master` melded one two commits that were there into one.
5. `git status` indicated I had 10 and 2  commits in local and remote branch respectively.
6. Did a `git pull --rebase`. 
7. Did `git push` and the Travis CI and Appveyor workflows seemed to have kicked off here in this PR.

Travis CI still seems unhappy, though I do see this passes locally when I run:

`tox -e py36`

And I can see commit `ceb0e409` via `git log` on that same topic branch.

**Comment by hackebrot** [comment]
Hi @terryjbates! 👋 

Not really sure what happened here.

Can you please try the following:

- ``git checkout master``
- ``git fetch upstream``
- ``git merge upstream/master``
- ``git push origin``

Now your master should be up-to-date both locally and your fork.

- ``git checkout tbates-pep257-742-exceptions``
- ``git rebase master``
- ``git log master..HEAD``

You should see two commits, both from you.

**Comment by terryjbates** [comment]
Thanks @hackebrot! I'm giving your commands a shot now. Thanks for helping get through merge hell. Looking back, I suspect I may not have pushed my local master to the master on GitHub.

**Pull request closed without merging**